### ISSUE TITLE
Page auto-description feature produces nonsensical descriptions

### ISSUE THREAD

**Issue by chalin** [google/docsy#968]
PR #662 introduced the auto-generation of page descriptions when none is provided in the page front matter. On many projects that I support, this behavior creates nonsensical descriptions, for example, the following is entirely blank:

> <img width="723" alt="image" src="https://user-images.githubusercontent.com/4140793/162575664-3cdf943c-1119-43c5-a262-fab90402ce6b.png">

Search engines can do a better job at auto generating a page's description.

This issue proposes that the page auto-description feature be opt-in for projects that want it, rather than the default.

/cc @LisaFC @emckean @geriom

**Comment by chalin** [comment]
When I first created this issue, I had suggested (via the title) that the page auto-description feature should be opt-in. Investigating this a bit more I've come to realized that Hugo's [internal templates] (such as for Twitter and Open Graph) consistently set meta-description elements using the same code, namely:

```nocode
{{ with .Description }}{{ . }}{{ else }}{{if .IsPage}}{{ .Summary }}{{ else }}{{ with .Site.Params.description }}{{ . }}{{ end }}{{ end }}{{ end }}
```

We should consider doing the same.

For internal template files, see https://github.com/gohugoio/hugo/tree/master/tpl/tplimpl/embedded/templates.

[internal templates]: https://gohugo.io/templates/internal/

**Comment by prepstarr** [comment]
My opinion if the opt in feature is obvious that could work but if for example someone see that page and the opt in feature was not obvious then they will be like well I want my meta headers as such... and they will look into ways of changing them so that it's what fits there needs but if they have this empty version it could serve as an empty template for them as well...

**Comment by chalin** [comment]
Closed by #1092
